src/value_type.rs

```diff
@@ -476,4 +476,41 @@ macro_rules! impl_primitive_type {
     )*}
 }
 
-impl_primitive_type!(Bool(bool), Char(char), String(String), U128(u128), I128(i128),);
+impl_primitive_type!(
+    Bool(bool as bool),
+    Char(char as char),
+    String(String as String),
+    U128(u128 as u128),
+    U128(u64 as u128),
+    U128(usize as u128),
+    U128(u32 as u128),
+    U128(u16 as u128),
+    U128(u8 as u128),
+    I128(i128 as i128),
+    I128(i64 as i128),
+    I128(isize as i128),
+    I128(i32 as i128),
+    I128(i16 as i128),
+    I128(i8 as i128),
+);
+
+// note regarding impl From<&str>:
+// a nicer generic `impl<K: Into<String>> From<K>` verson is not possible because it conflicts with the From<Bits> implementation above.
+
+impl From<&str> for Primitive {
```

Comment:
does `From<AsRef<str>>` work here instead?

Reply:
Sadly we cannot do this because of conflicting implementations. `scale_bits::Bits` might implement `AsRef<str>`, so there would be two conflicting ways of creating a `ValueDef` from `Bits`.